src/views/user/register.ejs

```diff
@@ -1,150 +1,301 @@
                         </div>
       
                       </div>
                     </div>
                     <div class="row">
-                        <div class="col-md-6 mb-4">
-        
-                          <div class="form-outline">
-                            <input type="text" id="street_and_num" class="form-control form-control-lg" />
-                            <label class="form-label" for="street_and_num">Street and number</label>
-                          </div>
-        
-                        </div>
-                        <div class="col-md-6 mb-4">
-        
-                          <div class="form-outline">
-                            <input type="text" id="city" class="form-control form-control-lg" />
-                            <label class="form-label" for="city">City</label>
-                          </div>
-        
+                      <div class="col-md-6 mb-4">
+      
+                        <div class="form-outline">
+                          <input type="text" id="zipcode"  required class="form-control form-control-lg" />
+                          <label class="form-label" for="zipcode">Zip-Code</label>
                         </div>
+      
                       </div>
-                      <div class="row">
-                        <div class="col-md-6 mb-4">
-        
-                          <div class="form-outline">
-                            <input type="text" id="zipcode" class="form-control form-control-lg" />
-                            <label class="form-label" for="zipcode">Zip-Code</label>
-                          </div>
-        
-                        </div>
-                        <div class="col-md-6 mb-4">
-        
-                          <div class="form-outline">
-                            <input type="text" id="country" class="form-control form-control-lg" />
-                            <label class="form-label" for="country">Country</label>
-                          </div>
-        
+                      <div class="col-md-6 mb-4">
+      
+                        <div class="form-outline">
+                          <input type="text" id="country" required class="form-control form-control-lg" />
+                          <label class="form-label" for="country">Country</label>
                         </div>
+      
                       </div>
-
-                    <div class="mt-4 pt-2">
-                      <input class="btn btn-primary btn-lg" type="submit" value="Submit" />
                     </div>
-      
-                  </form>
-                </div>
+
+                  <div class="mt-4 pt-2">
+                    <input class="btn btn-primary btn-lg" type="submit" id="submit_btn" value="Submit" />
+                  </div>
+    
+                </form>
               </div>
             </div>
           </div>
         </div>
-      </section>
+      </div>
+    </section>
 </main>
+<script>
```

Comment:
Czy to na pewno dobry pomysł żeby tak duży skrypt js był wewnątrz pliku html? Wygląda na to, że sporo logiki walidacyjnej powtarza się w `user_data_validation_rules.ts`.  Jesteśmy w stanie jakoś to ujednolicić i wyrzucić ten skrypt do osobnego pliku? A może byłoby z tym za dużo zachodu i lepiej zostawić jak jest?

Reply:
Stworzyłem nowy folder na skrypty dla frontendu public/js. Tam umieściłem ten skrypt. Ta logika walidacyjna powtarza się dlatego, że sprawdzam wszystkie dane na froncie i na serwerze.